keps/sig-security/NNNN-ambient-capabilities/README.md

```diff
@@ -0,0 +1,889 @@
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Changes to kubernetes API (https://pkg.go.dev/k8s.io/api/core/v1)
+
+Reusing the existing capabilities field in the securityContext might cause confusion and we propose adding a new field in the SecurityContext called ambientCapabilities. 
+
+
+```go
+type SecurityContext struct {
+	// The capabilities to add/drop when running containers.
+	// Defaults to the default set of capabilities granted by the container runtime.
+	// +optional
+	Capabilities *Capabilities `json:"capabilities,omitempty" protobuf:"bytes,1,opt,name=capabilities"`
+     // The ambient capabilities to add when running containers as non-root.
+     // +optional
+     AmbientCapabilities []Capability `json:"ambientCapabilities,omitempty" protobuf:"bytes,1,rep,name=ambientCapabilities,casttype=[]Capability"`
+
+// Rest of the struct omitted for brevity. You can find it at https://pkg.go.dev/k8s.io/api/core/v1#SecurityContext 
+```
+
+
+Notes about how ambientCapabilities would work:
+
+1. AmbientCapabilities can only be added.
+2. If the field is not supplied kubernetes will not add any capabilities to the ambient capabilities set by default.
+3. AmbientCapabilities field should only be set if the Pod is running as non-root. There are 2 ways to handle this:
+    1. Error if a Pod is running as root and has ambientCapabilities in the manifest.
+    2. Don’t error out but treat the ambientCapabilities field as a no-op.
+
+
+### Changes to runtime API (https://pkg.go.dev/k8s.io/cri-api/pkg/apis/runtime/v1alpha2)
+
+We propose adding a new field to the Capability struct
+
+```go
+type Capability struct {
+	// List of capabilities to add.
+	AddCapabilities []string `protobuf:"bytes,1,rep,name=add_capabilities,json=addCapabilities,proto3" json:"add_capabilities,omitempty"`
+	// List of capabilities to drop.
+	DropCapabilities     []string `protobuf:"bytes,2,rep,name=drop_capabilities,json=dropCapabilities,proto3" json:"drop_capabilities,omitempty"`
+	XXX_NoUnkeyedLiteral struct{} `json:"-"`
+	XXX_sizecache        int32    `json:"-"`
+          // List of ambient capabilities to add.
+          AddAmbientCapabilities []string
```

Comment:
Note: Even if we end up reusing the existing `Add` capabilities field in the k8s API, I agree that it should be kept separate at the CRI level.